ddocs/medic-api/updates.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * Given a document UUID and a message UUID update a task's state property on a
+ * document.
+ *
+ * PUT /db/_design/medic-api/_update/update_message_task/{docid}
+ *
+ * Parameters:
+ *
+ *   docid  document ID
+ *
+ * JSON request body paramters:
+ *
+ *   message_id     message uuid string
+ *   state          state value
+ *   state_message  message value [optional]
+ *
+ * Response Headers:
+ *
+ *   Content-Type  application/json
+ *   X-Couch-Id    Created/updated document’s ID
+ *   X-Couch-Update-Newrev  Created/updated document’s revision
+ *
+ * Status Codes:
+ *
+ *   201  Document was updated
+ *   404  Document not found
+ *   500  Internal Server Error – Query server error
+ *
+ * Response body on success:
+ *
+ *   {
+ *     "ok": true,
+ *   }
+ *
+ * Response body on failure includes error message:
+ *
+ *   {
+ *     "ok": false,
+ *     "error": "Your hair is not combed properly."
+ *   }
+ *
+ */
+
+exports.update_message_task = function(doc, request) {
+    var utils = require('medic-api-utils');
+    var data, msg;
+    var success = {
+      json: {
+        ok: true
+      }
+    };
+    var fail = function(msg, code) {
+      return [null, {
+        code: code || 500,
+        headers: {
+          'Content-Type' : 'application/json'
+        },
+        body: JSON.stringify({
+          ok: false,
+          error: msg
+        })
+      }];
+    };
+    if (!doc) {
+        return fail('Document not found: ' + doc, 404);
+    }
+    try {
+        data = JSON.parse(request.body);
+    } catch(e) {
+        return fail('Parsing JSON body failed');
+    }
+    if (!data.state) {
+        return fail('Value required for state.');
+    }
+    if (!data.message_id) {
+        return fail('Message id required');
+    }
+    msg = utils.getTask(data.message_id, doc);
+    if (!data.message_id || !msg) {
+        return fail('Message not found: ' + data.message_id);
+    }
+    utils.setTaskState(msg, data.state, data.details);
+    return [
+        doc,
+        success
+    ];
+};
```

Comment:
I don't like having this as a couchdb update function. Why not expose it as an express API (if needed externally) or include it in the npm utils (if only needed internally)? Then we could get rid of the ddoc altogether I think?